### ISSUE TITLE
Sistema de coding/quiz eventos

### ISSUE THREAD

**Issue by fernanduandrade** [he4rt/he4rt-bot-next#38]
Notei que alguns servers usam o bot para promover eventos para os membros, engajando os membros e dando alguma recompensa, achei uma ideia interessante de implementar no he4rt-bot.
O que eu pensei foi em implementar na api do he4rt-bot endpoints para termos um controle desses eventos e deixar dinâmico para criar novos
- Entidade para event, para ter uma descrição do evento, data que irá iniciar então dessa forma teria um cron job para verificar quando inicia o evento e quando chegar a data o bot anunciar que o evento está disponível para os membros participar, recompensas por participar
- Entidade de quiz, nela teríamos as perguntas e respostas e dicas de cada questão e ela teria relacionamento com o event 
- Recompensas seria xp como parte da gamefication e hcoins para todos que participarem e para os três primeiros alguma badge especial sugerido pelo @1pride

 Eu cheguei a fazer um mock de como seria o evento  mais tem muito a melhorar ainda

#### Início do evento quando o cron job identificar a data do evento, disparar a mensagem em um chat informando que os membros podem participar
<img width="713" alt="image" src="https://user-images.githubusercontent.com/58053397/209053926-a6aae1a9-d9bb-4a93-b479-36122a92b8f1.png">

#### Instruções e perguntas sendo recebidas pela dm do usuário
<img width="456" alt="image" src="https://user-images.githubusercontent.com/58053397/209054041-2a147737-5adf-47cc-8134-3a32b972fed6.png">

#### Após acertar todas as perguntas informar quem conseguiu no chat
<img width="526" alt="image" src="https://user-images.githubusercontent.com/58053397/209054117-1a25c0a5-e32b-4f7c-b469-71d7ede5a1c0.png">

**Comment by 1pride** [comment]
Então, eu vejo isso como um tipo de interação boa e se bem planejada e estruturada da para unir a comunidade e ver quem consegue contribuir em projetos futuros ou criar algo juntos, quem não conseguir resolver o problema poderia rolar tipo mentoria no dia para conseguirem finalizar o evento ganhando xp/coin e mais xp resolvendo problemas "comuns".

**Comment by Canhassi12** [comment]
Uma sugestão que eu tenho, que serveria para envolver mais outros membros da comunidade, é propor alguns projetos e meio que fazer desafios samanais ou mensais(A se pensar). E pegar todos os membros que estão dispostos, montar pequenos grupos dependendo da quantidade de gente e suas tecnologias. E em determinado tempo, na reunião ter uma avaliação dos projetos, ter algumas referencias backend, frontend e debater de acordo com o que foi aprensentado com a galera. É mais uma forma de interação e integrar iniciantes na área a fazer projetos em grupo, pode ser algo importante para conhecer o mercado.

**Comment by Luisnadachi** [comment]
Acho válido os pontos que o @1pride e o @Canhassi12 trouxeram, só pra complementar eles, acho que é bem interessante que isso pode abrir portas para jogos de coding mais avançados até, então servira como um bom teste começar com quiz e desafios de codigo agora no inicio que dando certo no futuro podemos evoluir isso.

**Comment by fernanduandrade** [comment]
> Acho válido os pontos que o @1pride e o @Canhassi12 trouxeram, só pra complementar eles, acho que é bem interessante que isso pode abrir portas para jogos de coding mais avançados até, então servira como um bom teste começar com quiz e desafios de codigo agora no inicio que dando certo no futuro podemos evoluir isso.

concordo, não sendo algo complexo de início apenas para iniciar e coletar feedback dos participantes e entender no que podemos melhorar

**Comment by Novout** [comment]
@fernanduandrade há alguma *branch* com isto implementado?

**Comment by fernanduandrade** [comment]
salve @Novout  😎! Eu fiz somente algumas alterações local onde tinha usado os dados de um json para fazer o fluxo do evento, cheguei a apresentar para o @1pride. No momento estou procurando por uma API 0800 que compila código e retorne um output para a parte de coding, encontrei algumas porém são pagas rsrs e estou modelando o banco e criando um fluxo dos endpoints

**Comment by Novout** [comment]
#83
